src/Service/Client/ServiceClient.php

```diff
@@ -74,12 +75,29 @@ public function request(
         array $scopes = []
     ): ResponseInterface {
         try {
-            $options = array_merge_recursive($options, [
-                'headers' => ['Authorization' => sprintf('Bearer %s', $this->getAccessToken($registration, $scopes))]
-            ]);
+            $options = array_merge_recursive(
+                $options,
+                [
+                    'headers' => [
+                        'Authorization' => sprintf('Bearer %s', $this->getAccessToken($registration, $scopes))
+                    ]
+                ]
+            );
+
+            try {
+                return $this->client->request($method, $uri, $options);
+            } catch (ClientException $exception) {
+                if ($exception->getResponse()->getStatusCode() === 401) {
```

Comment:
You've mentioned it's possible to inject a guzzle client from the user-space code.
So, what'll happen if they inject it with `['http_errors' => false]`? :)

WDYT of checking the response even if there's no `ClientException` thrown?

Reply:
I simply enforced `'http_errors' => true` at forst request, to be sure I'll catch in case of error.